packages/text-field/index.js

```diff
@@ -0,0 +1,367 @@
+      ]);
+    }
+    return textField;
+  }
+
+  inputProps = (props) => {
+    const {foundationValue} = this.state;
+    return Object.assign({}, props, {
+      foundationValue,
+      foundation: this.foundation_,
+      updateFocus: (isFocused) => this.setState({isFocused}),
+      handleValueChange: (value) => this.setState({value}),
+      // These are callbacks for Input, which set validity.badInput &
+      // validity.valid.
+      setBadInputHandler: (getBadInput) => this.getBadInput = getBadInput,
+      setIsValidHandler: (getIsValid) => this.getIsValid = getIsValid,
+      setDisabled: (disabled) => this.setState({disabled}),
+      setInputId: (id) => this.setState({inputId: id}),
+    });
+  }
+
+  renderInput() {
+    const child = React.Children.only(this.props.children);
+    const props = this.inputProps(child.props);
+    return React.cloneElement(child, props);
+  }
+
+  renderLabel() {
+    const {label, floatingLabelClassName} = this.props;
+    const {inputId} = this.state;
+    return (
+      <FloatingLabel
+        className={floatingLabelClassName}
+        float={this.state.shouldFloatLabel}
+        handleWidthChange={
+          (labelWidth) => this.setState({labelWidth})}
+        ref={this.floatingLabelElement}
+        htmlFor={inputId}
+      >
+        {label}
+      </FloatingLabel>
+    );
+  }
+
+  renderLineRipple() {
+    const {lineRippleClassName} = this.props;
+    const {activeLineRipple, lineRippleCenter} = this.state;
+    return (
+      <LineRipple
+        rippleCenter={lineRippleCenter}
+        className={lineRippleClassName}
+        active={activeLineRipple}
+      />
+    );
+  }
+
+  renderNotchedOutline() {
+    const {notchedOutlineClassName} = this.props;
+    const {shouldNotchOutline, labelWidth} = this.state;
+    return (
+      <NotchedOutline
+        className={notchedOutlineClassName}
+        isRtl={this.getIsRtl()}
+        notch={shouldNotchOutline}
+        notchWidth={labelWidth}
+      />
+    );
+  }
+
+  renderHelperText() {
+    const {helperTextAriaHidden, helperTextIsValidation, helperText,
+      helperTextClassName, helperTextPersistent, helperTextRole} = this.props;
+    const {isValid, helperTextShowToScreenReader} = this.state;
+    return (
+      <HelperText
+        className={helperTextClassName}
+        showToScreenReader={helperTextShowToScreenReader}
+        isValid={isValid}
+        role={helperTextRole}
+        key='text-field-helper-text'
+        aria-hidden={helperTextAriaHidden}
+        persistent={helperTextPersistent}
+        isValidation={helperTextIsValidation}
+      >
+        {helperText}
+      </HelperText>
+    );
+  }
+
+  renderIcon(icon) {
+    const {disabled} = this.state;
+    // Toggling disabled will trigger icon.foundation.setDisabled()
+    return (
+      <Icon disabled={disabled}>
+        {icon}
+      </Icon>
+    );
+  }
+}
+
+TextField.propTypes = {
+  'box': PropTypes.bool,
```

Comment:
Remove the single quotes from these unless they're necessary.

Reply:
I don't like them either, but they are necessary for `children.props`

Reply:
Gotcha. We can leave them then.